support/fc_sort.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python3
+
+"""Sort file context definitions
+
+The original setfiles sorting algorithm did not take into
+account regular expression specificity. With the current
+strict and targeted policies this is not an issue because
+the file contexts are partially hand sorted and concatenated
+in the right order so that the matches are generally correct.
+The way reference policy and loadable policy modules handle
+file contexts makes them come out in an unpredictable order
+and therefore setfiles (or this standalone tool) need to sort
+the regular expressions in a deterministic and stable way.
+"""
+
+import sys
+import argparse
+from pathlib import Path
+import re
+import functools
+
+
+class FileContext():
+    """ Container class for file context defintions
+    """
+
+    def __init__(self, path, file_type, context):
+        """ Constructor
+        """
+        self.path = path
+        self.file_type = file_type
+        self.context = context
+
+        self.compute_diffdata()
+
+    def compute_diffdata(self):
+        """ Compute the interal values needed for comparing two file context definitions
+        """
+
+        self.meta = False
+        self.stem_len = 0
+        self.str_len = 0
+
+        skip_escaped = False
+
+        for char in self.path:
+            if skip_escaped:
+                skip_escaped = False
+                continue
+
+            if char in ('.', '^', '$', '?', '*', '+', '|', '[', '(', '{',):
+                self.meta = True
+            if char == '\\':
+                skip_escaped = True
```

Comment:
It might be simpler to use `str.find()`. Then when you get an index, you do index-1 to see if it is escaped. That will simultaneously allow you to determine the stem length too. If you enumerate() too, you can exit the loop once the charater count > stem_len.

I also don't think that the `self.meta` is really needed. `If stem_len:` would be true if there are metacharacters.

Reply:
I think that would complicate the logic: one would need to rescan with `str.find()` if `index-1` is an escape character

Reply:
Ah yes, you're right of course.

Reply:
Ok.